google/src/main/scala/org/broadinstitute/dsde/workbench/google/HttpGoogleStorageDAO.scala

```
@@ -65,9 +65,13 @@ class HttpGoogleStorageDAO(appName: String,
     // If `recurse` is true, first delete all objects in the bucket
     val deleteObjectsFuture = if (recurse) {
       val listObjectsRequest = storage.objects().list(bucketName.value)
-      retryWhen500orGoogleError(() => executeGoogleRequest(listObjectsRequest)).flatMap { objects =>
+      retryWithRecoverWhen500orGoogleError { () =>
+        Option(executeGoogleRequest(listObjectsRequest))
```

Isn't this going to swallow all exceptions?

---

It should just recover from 404s, making this `Future` a no-op. Other errors should result in a failed Future. We recover from 404s below in `deleteBucketRequest` as well.

I hit this in Leo when resuming a paused cluster -- it triggers this code to delete the init bucket, but the init bucket has already been deleted when the cluster started up the first time. So we don't want this method to fail on 404 errors.

---

I mean the `Option()` here.  The recover for 404 looks fine.

---

Oh sorry. `Option(_)` shouldn't swallow exceptions (unlike `Try`), but it should handle null responses. (I never know when Google APIs are going to return null)

---

Ah!  Did not realize that.  Looks good.